pyiceberg/table/__init__.py

```diff
@@ -871,6 +924,12 @@ def sort_orders(self) -> Dict[int, SortOrder]:
         """Return a dict of the sort orders of this table."""
         return {sort_order.order_id: sort_order for sort_order in self.metadata.sort_orders}
 
+    def last_partition_id(self) -> int:
+        """Return the highest assigned partition field ID across all specs for the table or 999 if there is no spec."""
+        if self.metadata.last_partition_id:
+            return self.metadata.last_partition_id
+        return PARTITION_FIELD_ID_START - 1
```

Comment:
We can't just return self.metadata.last_partition_id because that is technically an optional type, but in practice will always be set due to our model setting the value. I don't think we should change that because in V1 it is technically optional, that's why we have our model set the value after words.

Reply:
In PyIceberg we treat all the tables internally as V2 tables. From the spec, it should start at 1000:

<img width="702" alt="image" src="https://github.com/apache/iceberg-python/assets/1134248/e750ccea-20d7-4c37-acae-6a09c8feed4b">

Reply:
https://github.com/apache/iceberg-python/pull/245/files#r1479266072 discussion on this, going to resolve this thread.